### PR TITLE
feat(component): add neural network components

### PR DESCRIPTION
**Pull request by sahandsafizadeh**
- Feat!: add `component` tree
A ***component***, is a separate computational functionality of a neural network. They are categorized into 5 major packages:
1. `initializers`: including **Full**, **Uniform**, **Normal**, **He Uniform**, **He Normal**, **Xavier Uniform** and **Xavier Normal** methods for initializing weights.
2. `layers`: including *activation functions* **Sigmoid**, **Softmax**, **Tanh**, **Relu**, **Leaky Relu** and *trainable modules* such as **Fully Connected**. These components build up a network and tensors pass through them.
3. `losses`: including **Mean Squared Error**, **Binary Cross Entropy** and **Cross Entropy** methods to back-propagate.
4. `metrics`: including **Accuracy** method for evaluating the model.
5. `optimizers`: including **Stochastic Gradient Descent** method for updating network weights.

### CODE REVIEW

**Review by sahandsafizadeh — COMMENTED** [review]
approved

**Review by sahandsafizadeh — COMMENTED** [review]
approved
